cmd/kubeadm/app/cmd/phases/kubeconfig.go

```diff
@@ -74,7 +76,8 @@ func NewCmdClientCerts(out io.Writer) *cobra.Command {
 }
 
 func addCommonFlags(cmd *cobra.Command, config *kubeconfigphase.BuildConfigProperties) {
-	cmd.Flags().StringVar(&config.CertDir, "cert-dir", kubeadmapiext.DefaultCertificatesDir, "The path to the directory where the certificates are.")
+	certificatesDir := filepath.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeadmconstants.CertificatesDirName)
```

Comment:
One important thing to realize is that KubernetesDir and CertificatesDir should be able to be in two different roots, they should be able to be decoupled.
i.e. KubernetesDir could be in /etc/kubernetes (the default) but CertificatesDir /var/lib/kube-certificates